contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/util/SimpleHttp.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.http.util;
+
+import okhttp3.Cache;
+import okhttp3.Credentials;
+import okhttp3.FormBody;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.OkHttpClient.Builder;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.drill.exec.ExecConstants;
+import org.apache.drill.exec.ops.FragmentContext;
+import org.apache.drill.exec.store.http.HttpAPIConfig;
+import org.apache.drill.exec.store.http.HttpStoragePluginConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+
+/**
+ * This class performs the actual HTTP requests for the HTTP Storage Plugin. The core method is the getInputStream()
+ * method which accepts a url and opens an InputStream with that URL's contents.
+ */
+public class SimpleHttp {
+  private static final Logger logger = LoggerFactory.getLogger(SimpleHttp.class);
+
+  private final OkHttpClient client;
+
+  private final HttpStoragePluginConfig config;
+
+  private final FragmentContext context;
+
+  private final HttpAPIConfig apiConfig;
+
+  public SimpleHttp(HttpStoragePluginConfig config, FragmentContext context, String connectionName) {
+    this.config = config;
+    this.context = context;
+    this.apiConfig = config.connections().get(connectionName);
+    client = setupServer();
+  }
+
+
+
+  public InputStream getInputStream(String urlStr) {
+    Request.Builder requestBuilder;
+
+    // The configuration does not allow for any other request types other than POST and GET.
+    if (apiConfig.method().equals("get")) {

```

Comment:
Here we assume the user knows to use lower case (`"get"`) for the method. This is not entirely obvious since it is very common to refer to methods by their upper-case names: `GET`, `POST`, etc. Maybe `equalsIgnoreCase()`?

Reply:
Also, what happens if the user leaves this field empty or null? Should these attributes be checked at some point? Perhaps in the `@JsonCreator` so we fail early, rather than at run time?

Reply:
In the HTTPApiConfig class, it validates this field and converts to lowercase. The plugin accepts only `get` and `post` requests.  If the field is `null` or some other unsupported method, the plugin falls back to `get`.

[1]: https://github.com/apache/drill/blob/3190a0d43cf7959fa6f17f75e966dc1a924a6c1b/contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpAPIConfig.java#L56-L60